.github/workflows/tests.yml

```diff
@@ -627,6 +627,9 @@ jobs:
     strategy:
       matrix:
         os: [windows-latest, ubuntu-latest]
+        folder:
+          - "rest_api"
+          - "ui"        
```

Comment:
I get the idea behind this change, but I think it's overkill. The UI has like two unit tests only... it's not worth to pull up a separate GH runner for those. The setup time is way longer than the tests it executes. Can we revert this?

(Same for the follow-up changes below)

Reply:
I see only now your comment above, but anyway I think this is not a sustainable fix. What errors did you observe?

Reply:
I understand your point, so I reverted this change. 

There was an error related to test import ([complete logs](https://github.com/deepset-ai/haystack/actions/runs/3259039230/jobs/5351589109)):

> __________________ ERROR collecting ui/test/test_ui_utils.py ___________________
> ImportError while importing test module '/home/runner/work/haystack/haystack/ui/test/test_ui_utils.py'.
> Hint: make sure your test modules/packages have valid Python names.
> Traceback:
> /opt/hostedtoolcache/Python/3.7.14/x64/lib/python3.7/importlib/__init__.py:127: in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
> E   ModuleNotFoundError: No module named 'test.test_ui_utils'

I read something about this in [pytest docs](https://docs.pytest.org/en/7.1.x/example/pythoncollection.html), but is an intricate topic.
Anyway, now I solved the issue, by calling `pytest ${{ env.PYTEST_PARAMS }} --import-mode=importlib rest_api/ ui/`
WDYT? Is this solution acceptable?

Reply:
This last approach looks much better, thanks :blush: Still a bit weird but I'm not too surprised, given that we're running tests from two separate packages in the same pytest line.

Last minute thought: have you tried to simply run the tests on two separate pytest calls? I would guess that pytest would have way less trouble with that. If you tried this and it also doesn't work, then let's keep your solution.
```yaml
    - name: Run tests
      run: |
        pytest ${{ env.PYTEST_PARAMS }} rest_api/
        pytest ${{ env.PYTEST_PARAMS }} ui/
```

Reply:
Let me try your proposal...

Reply:
Your proposal works and is very clean!